Doppler.Sap/Mappers/BusinessPartnerMapper.cs

```diff
@@ -21,31 +21,68 @@ public static string MapDopplerUserIdToSapBusinessPartnerId(int id, int planType
 
         public static SapBusinessPartner MapDopplerUserToSapBusinessPartner(DopplerUserDto dopplerUser, string cardCode, SapBusinessPartner fatherBusinessPartner)
         {
-            var newBusinessPartner = new SapBusinessPartner
+            var newBusinessPartner = new SapBusinessPartner();
+
+            var countryCode = Enum.Parse<CountryCodeEnum>(dopplerUser.BillingCountryCode);
+            var sapBillStateId = string.Empty;
+            var sapShipStateId = string.Empty;
+
+            var sapValues = countryCode == CountryCodeEnum.AR
+                ? new
+                {
+                    PayTermsGrpCode = dopplerUser.PaymentMethod == (int)PaymentMethodEnum.MP ? (int)PayTermsGroupEnum.MP : (int)PayTermsGroupEnum.DEFAULT,
+                    Currency = "##",
+                    FederalTaxId = dopplerUser.FederalTaxID.Replace("-", ""),
+                    U_B1SYS_VATCtg = dopplerUser.IdConsumerType.HasValue ?
+                        (Dictionary.ConsumerTypesDictionary.TryGetValue(dopplerUser.IdConsumerType, out string consumerType) ? consumerType : "CF")
+                        : "CF",
+                    U_B1SYS_FiscIdType = dopplerUser.FederalTaxType == "DNI" ? "96" : (dopplerUser.FederalTaxType == "CUIT" ? "80" : "99"),
+                    EmailGroupCode = "Facturacion",
+                    SalesPersonCode = (dopplerUser.IsInbound.HasValue ? (dopplerUser.IsInbound.GetValueOrDefault() ? 1 : 2) : 3),
+                    Indicator = "FC",
+                    DunningTerm = "ReclamoVto",
+                    BillToState = (dopplerUser.BillingStateId.HasValue ?
+                                (Dictionary.StatesDictionary.TryGetValue(dopplerUser.BillingStateId.Value, out sapBillStateId) ? sapBillStateId : "99") : "99"),
+                    ShipToState = (dopplerUser.BillingStateId.HasValue ?
+                                (Dictionary.StatesDictionary.TryGetValue(dopplerUser.BillingStateId.Value, out sapShipStateId) ? sapShipStateId : "99") : "99")
+                }
+                : new
+                {
+                    PayTermsGrpCode = -1,
+                    Currency = "$",
+                    FederalTaxId = dopplerUser.FederalTaxID,
+                    U_B1SYS_VATCtg = (string)null,
+                    U_B1SYS_FiscIdType = (string)null,
+                    EmailGroupCode = "Billing",
+                    SalesPersonCode = (dopplerUser.IsInbound.HasValue ? (dopplerUser.IsInbound.GetValueOrDefault() ? 1 : 2) : 3),
+                    Indicator = (string)null,
+                    DunningTerm = (string)null,
+                    BillToState = (Dictionary.StatesDictionary.TryGetValue(dopplerUser.BillingStateId.Value, out sapBillStateId) ? sapBillStateId : "99"),
+                    ShipToState = (Dictionary.StatesDictionary.TryGetValue(dopplerUser.BillingStateId.Value, out sapShipStateId) ? sapShipStateId : "99")
+                };
```

Comment:
It would be nice using a factory for the mappers, as you are doing with the handlers, in that way you could declare them in the appsettings and avoid this conditionals.